ow_lander/scripts/all_action_trajectories.py

```diff
@@ -0,0 +1,853 @@
+ #!/usr/bin/env python2
+
+# The Notices and Disclaimers for Ocean Worlds Autonomy Testbed for Exploration
+# Research and Simulation can be found in README.md in the root directory of
+# this repository.
+
+import rospy
+import math
+import constants
+import copy
+from tf.transformations import quaternion_from_euler
+from tf.transformations import euler_from_quaternion
+from utils import is_shou_yaw_goal_in_range
+from moveit_msgs.msg import PositionConstraint
+from geometry_msgs.msg import Quaternion
+from shape_msgs.msg import SolidPrimitive
+from moveit_msgs.msg import RobotTrajectory
+from trajectory_msgs.msg import JointTrajectory
+from trajectory_msgs.msg import JointTrajectoryPoint
+from std_msgs.msg import Header
+
+
+
+deliver_sample_traj = RobotTrajectory()
+
+def calculate_joint_state_end_pose_from_plan_arm (robot, plan, move_arm, moveit_fk):
+  ''' 
+  calculate the end pose (position and orientation), joint states and robot states
+  from the current plan
+  inputs:  current plan, robot, arm interface, and moveit forward kinematics object
+  outputs: goal_pose, robot state and joint states at end of the plan
+  '''  
+  #joint_names: [j_shou_yaw, j_shou_pitch, j_prox_pitch, j_dist_pitch, j_hand_yaw, j_scoop_yaw]
+  #robot full state name: [j_ant_pan, j_ant_tilt, j_shou_yaw, j_shou_pitch, j_prox_pitch, j_dist_pitch, j_hand_yaw,
+  #j_grinder, j_scoop_yaw]
```

Comment:
Is this comment useful and does it need to be repeated in many functions?

Reply:
I think it is useful... there is a difference between the joints and the full robot state.. and the order of the names differ based on the control group chosen..... thus it is a good idea to state the joint names.....